api/tests/test_gdpr/test_gdpr_api.py

```diff
@@ -0,0 +1,159 @@
+import datetime
+import uuid
+
+import requests_mock
+from assertpy import assert_that
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.test import override_settings
+from django.urls import reverse
+from helusers.settings import api_token_auth_settings
+from jose import jwt
+from rest_framework.test import APITestCase
+
+from api.tests.test_gdpr.gdpr_key import rsa_key
+from applications.tests.factories import AddressFactory, ApplicationFactory
+from reservations.tests.factories import ReservationFactory
+
+User = get_user_model()
+
+
+@override_settings(
+    OIDC_API_TOKEN_AUTH={
+        "AUDIENCE": "test_audience",
+        "API_SCOPE_PREFIX": "testprefix",
+        "ISSUER": "http://localhost/openid",
+        "TOKEN_AUTH_REQUIRE_SCOPE_PREFIX": True,
+    },
+    GDPR_API_QUERY_SCOPE="testprefix.gdprquery",
+    GDPR_API_DELETE_SCOPE="testprefix.gdprdelete",
+)
+class TilavarauspalveluGDPRAPIViewTestCase(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.user = User.objects.create(
+            username="regjoe",
+            first_name="joe",
+            last_name="regular",
+            email="joe.regular@foo.com",
+        )
+        ReservationFactory(user=cls.user)
+        billing_address = AddressFactory()
+        ApplicationFactory(user=cls.user, billing_address=billing_address)
+        cls.url = reverse("gdpr_v1", kwargs={"uuid": cls.user.uuid})
+
+    def setUp(self) -> None:
```

Comment:
Ahaa! This resets the user data that might change in the tests?

Can you add a clarifying comment to reduce confusion in the future :pleading_face:

Reply:
Yes :+1:  It could also create the whole data here as well but the class method (setUpTestData) is better usually. 
`setUp` method is useful for "setupping" each test since `setUpTestData` is a class method which is called when the `TestCase` class is created => data changes between tests are possible => `setUp` is run per test.